### PR TITLE
add dependencies of repology

### PR DESCRIPTION
**Pull request by ryantm**
###### Motivation for this change

I want to run Repology so I can potentially modify it help with https://github.com/NixOS/nixpkgs/issues/41029

###### Things done

- [x] built all these on NixOS
- [x] ran Repology's `python repology-update.py -f -u  -p nix` in a nix-shell using these dependencies

### CODE REVIEW

File: pkgs/development/python-modules/rubymarshal/default.nix [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+buildPythonPackage rec {
+  pname = "rubymarshal";
+  version = "1.0.3";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "131lbc18s3rlmby2dpbvi4msz13gqw6xvx067mh4zcx9npygn9r2";
+  };
+
+  buildInputs = [ hypothesis ];
```
Comment (dotlambda) [inline pkgs/development/python-modules/rubymarshal/default.nix]: `propagatedBuildInputs`

File: pkgs/development/libraries/libversion/default.nix [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+  name = "libversion-${version}";
+
+  src = fetchFromGitHub {
+    owner = "repology";
+    repo = "libversion";
+    rev = "${version}";
+    sha256 = "0krhfycva3l4rhac5kx6x1a6fad594i9i77vy52rwn37j62bm601";
+  };
+
+  buildInputs = [ cmake ];
```
Comment (dotlambda) [inline pkgs/development/libraries/libversion/default.nix]: `nativeBuildInputs`

File: pkgs/development/libraries/libversion/default.nix [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+let
+  version = "2.6.0";
+in
+stdenv.mkDerivation {
+  name = "libversion-${version}";
+
+  src = fetchFromGitHub {
+    owner = "repology";
+    repo = "libversion";
+    rev = "${version}";
```
Comment (dotlambda) [inline pkgs/development/libraries/libversion/default.nix]: `rev = version`

**Review by dotlambda — CHANGES_REQUESTED** [review]

**Comment by ryantm** [comment]
@dotlambda Thanks for the feedback; I made those changes.

**Comment by dotlambda** [comment]
> ## Requirements
> - Python 3.6+

(https://github.com/repology/py-libversion#requirements)

**Comment by ryantm** [comment]
@GrahamcOfBorg build python2.pkgs.libversion python3.pkgs.libversion python2.pkgs.rubymarshal python3.pkgs.rubymarshal

File: pkgs/development/python-modules/libversion/default.nix [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "18hhn7b7458lybs8z8ckh0idm7a2g4c4b5v2p9rr0lb618rchvds";
+  };
+
+  nativeBuildInputs = [ pkgconfig ];
+  buildInputs = [ libversion ];
+
+  doCheck = true;
```
Comment (dotlambda) [inline pkgs/development/python-modules/libversion/default.nix]: That's set automatically for python packages. I meant `libversion`, not `pythonPackages.libversion`

Comment (ryantm) [reply pkgs/development/python-modules/libversion/default.nix]: Oh, okay! For `libversion` says: `make: *** No rule to make target 'check'.  Stop.`

Comment (dotlambda) [reply pkgs/development/python-modules/libversion/default.nix]: `checkTarget = "test"`

**Comment by dotlambda** [comment]
@ryantm Now please squash the last two commits into the appropriate ones and we're good to go.

@GrahamcOfBorg build python3.pkgs.libversion, python2.pkgs.rubymarshal, python3.pkgs.rubymarshal

**Pull request merged**